File: Program/1_Array/Array Standard problem/easy/6_method_1_rotate_the_array_by_d_position_to_the_left.c
//method 1 : Using temp array

#include <stdio.h>

void d_rotate(int arr[], int size)
{
    int pos = 3;
    int temp[size], j = 0;

    for (int i = pos - 1; i < size; i++)
    {
        temp[j] = arr[i];
        j++;
    }

    for (int i = 0; i < pos-1; i++)
    {
        temp[j] = arr[i];
        j++;
    }

    for (int i = 0; i < size; i++)
    {
        printf("%d |", temp[i]);
    }
}

void main()
{
    int arr[] = {7, 5, 4, 11, 2, 10, 13,45,101};
    int size = sizeof(arr) / sizeof(int);

    printf("The original array : ");
    for (int i = 0; i < size; i++)
    {
        printf("%d |", arr[i]);
    }
    printf("\n");
    printf("d rotate array : ");

    d_rotate(arr, size);
}

File: Program/1_Array/Array Standard problem/easy/6_method_1_rotate_the_array_by_d_position_to_the_left.py
def rotate_arr(arr, pos):
    
    size = len(arr)
    if pos > size:
        return -1
    temp = arr.copy()
    # temp = []
    # temp.extend(arr)
    count = 0
    for i in range(pos-1, size):
        temp[count] = arr[i]
        count+=1

    for i in range(pos-1):
        temp[count] = arr[i]
        count +=1

    return temp


l = [7, 5, 4, 11, 2, 10, 13,45,101]
pos = 3
print(rotate_arr(l,pos))